### DISCUSSION TITLE
Adding text to PhotoDemon

### DISCUSSION THREAD

**Discussion by dcannuck** [tannerhelland/PhotoDemon#445] [Q&A]
Hi,
Anybody have any tips on adding text in PhotoDemon9?
I just wanted to do some simple things like adding words to photos, and I am finding it extremely frustrating.
I type a few words; then change the font to bold or italic text, and all the previously typed words revert to bold or italic. How do I switch between formats, and preserve the previously chosen font?      I wanted a sentence that **LOOKED **LIKE this, but I can't figure out a way to do it.****

**Comment by tannerhelland** [comment]
Hello @dcannuck .  Thank you for your question.  I'm so sorry this has been frustrating for you!

Like most free photo editors, text layers in PhotoDemon only support a single uniform text style for an entire text layer (including font face, size, style, etc).  If you need to mix-and-match font settings, you will need a separate text layer for each style.

Unfortunately, this is a limitation of the built-in Windows edit box used for editing text.  PhotoDemon leans heavily on built-in Windows features like this which is how it maintains such a small app size.

If you need to use multiple styles inside a single text layer, you might be better served by a much larger photo editor like [GIMP](https://www.gimp.org/).  

Thanks again for your feedback, and good luck with your project.

**Reply by dcannuck** [reply]
Hi Tanner,

Thanks for getting back to me.
 
I'm not that up to date with the latest photo editors that are out there for free. Wasn't aware of these limitations regarding text.  I have been using an old Photoshop Elements program for most of my editing needs, but found that if I added text with that program, it didn't appear clear; rather jagged edges for any font.
I still like PhotoDemon as a program. I like that it's portable (I installed it on a stick), and it has some interesting filters and effects.
Gimp looks interesting, may give it a try.
